src/components/resizable_container/resizable_container.tsx

```diff
@@ -68,9 +69,21 @@ export interface EuiResizableContainerProps
    */
   onPanelWidthChange?: ({}: { [key: string]: number }) => any;
   onToggleCollapsed?: ToggleCollapseCallback;
+  /**
+   * Called when resizing starts
+   */
+  onResizeStart?: (trigger: 'pointer' | 'key') => any;
+  /**
+   * Called when resizing ends
+   */
+  onResizeEnd?: () => any;
   style?: CSSProperties;
 }
 
+type ResizeTrigger = Parameters<
+  NonNullable<EuiResizableContainerProps['onResizeStart']>
+>[0];
+
```

Comment:
It might be a tiny bit easier to read to define this as
```suggestion
type ResizeTrigger = 'pointer' | 'key';
```

and have the `onResizeStart` type def above call this instead of the other way around, e.g.
```
onResizeStart?: (trigger: ResizeTrigger) => void
```

```

Reply:
Yeah this was a pretty ugly type 😅 My concern here was referencing the internal `ResizeTrigger` type in an exported callback, mainly since it wasn't clear in the props tab in the docs. But I saw what others were doing to get around this with `keyof typeof`, so I tried a similar thing for `ResizeTrigger` that seems to work and is cleaner than what I had before I think. Also moved it to `types.ts` while I was at it.